Sources/HummingbirdHTTP2/TLSChannelConfiguration.swift

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2025 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOSSL
+
+/// TLSChannel configuration
+public struct TLSChannelConfiguration: Sendable {
+    public typealias CustomVerificationCallback = @Sendable ([NIOSSLCertificate], EventLoopPromise<NIOSSLVerificationResult>) -> Void
```

Comment:
Should we expose an API here that has a signature of `@Sendable ([NIOSSLCertificate]) async throws -> NIOSSLVerificationResult`?

Reply:
Had a look at this and that isn't what is expected of this function. If you check the function it sends in a promise, kicks off the custom verification process and returns immediately. Changing it to async/await means it will wait until the custom verification has completed. I could kick off an unstructured task inside the function. Not sure that's ideal though.

Reply:
In actual fact given how promise.completeWithTask works it doesnt really make much difference if it is an unstructured task. I have added an async verification callback, but left the EL version in as well